applications: nrf_desktop: Avoid using Zephyr Bluetooth role Kconfigs (#9886)
nrfconnect/sdk-nrf: applications/nrf_desktop/doc/hid_forward.rst

Complete the following steps to configure the module:

1. Complete the basic Bluetooth configuration, as described in :ref:`nrf_desktop_bluetooth_guide`.
   Make sure that both :ref:`CONFIG_DESKTOP_ROLE_HID_DONGLE <config_desktop_app_options>` and :ref:`CONFIG_DESKTOP_BT_CENTRAL <config_desktop_app_options>` are enabled.
#. Enable and configure the :ref:`hogp_readme` (:kconfig:option:`CONFIG_BT_HOGP`).
   An nRF Desktop dongle does not generate its own HID input reports.

>    The dongle uses |hid_forward| to forward the HID reports that are received by the HID service client from Bluetooth connected peripherals.

Reviewer: ```suggestion
   The dongle uses |hid_forward| to forward the HID reports. 
   The reports are then received by the HID service client from the peripherals connected over Bluetooth.
```

   .. note::
       Make sure to define the maximum number of supported HID reports (:kconfig:option:`CONFIG_BT_HOGP_REPORTS_MAX`).

> #. The :ref:`CONFIG_DESKTOP_HID_FORWARD_ENABLE <config_desktop_app_options>` option enables the |hid_forward|.
>    The option is automatically selected by the :ref:`CONFIG_DESKTOP_ROLE_HID_DONGLE <config_desktop_app_options>`.

Reviewer: ```suggestion
#. The :ref:`CONFIG_DESKTOP_HID_FORWARD_ENABLE <config_desktop_app_options>` option enables the |hid_forward| and is automatically selected by the :ref:`CONFIG_DESKTOP_ROLE_HID_DONGLE <config_desktop_app_options>` option.
```

> #. nRF Desktop dongle may forward either mouse or keyboard boot reports.

Reviewer: ```suggestion
#. nRF Desktop dongle can forward either mouse or keyboard boot reports.
```
